develop/jsapi/search_with_widget/lab.md

```diff
@@ -0,0 +1,89 @@
+
+    ```javascript
+    require([
+      "esri/Map",
+      "esri/views/MapView",
+      "esri/widgets/Search",
+      "esri/layers/FeatureLayer",
+      "dojo/domReady!"
+    ], function(Map, MapView, Search, FeatureLayer) {
+      ... 
+    ```
+
+3. Create the Search widget and add it to the `top-right` of the UI.
+
+    ```javascript
+    ], function(Map, MapView, Search, FeatureLayer) {
+
+      var map = new Map({
+        basemap: "dark-gray"
+      });
+
+      var view = new MapView({
+        container: "viewDiv",
+        map: map,
+        center: [-122.68, 45.52], // lon, lat
+        zoom: 10
+      });
+
+      var searchWidget = new Search({
+        view: view,
+        allPlaceholder: "Neighborhood e.g. Downtown"
+      });
+
+      searchWidget.startup();
+
+      view.ui.add(searchWidget, {
+        position: "top-right"
+      });
+    ```
+
+    At this point, the map will allow you to search against the default ArcGIS Online Geocoding Service. Give it a go. You can enter an address or point of interest (like `Providence Park` or `PDX`) or a geography (like `Oregon` or `USA`).
+
+4. Now configure the Search Widget to just search against the Neighborhoods Feature Service. Insert the following code directly before the `search.startup()` line added above.
+
+    ```javascript
+      var searchWidget = new Search({
+        view: view,
+        allPlaceholder: "Neighborhood e.g. Downtown"
+      });
+
+      var sources = [];
+      
+      sources.push({
+        featureLayer: new FeatureLayer("http://services.arcgis.com/uCXeTVveQzP4IIcx/arcgis/rest/services/PDX_Neighborhoods_Styled/FeatureServer/0"),
+        name: "Neighborhood Search",
+        searchFields: ["NAME"],
+        displayField: "NAME",
+        exactMatch: false,
+        outFields: ["NAME","AVGHINC_CY","MEDAGE_CY","TOTPOP_CY"],
+        placeholder: "Neighborhood e.g. Downtown",
+        // Create a PopupTemplate to format data
+        popupTemplate: {
+          title: "{NAME}",
+          content: "Median Age: {MEDAGE_CY}</br>Average Household Income: {AVGHINC_CY}</br> Population: {TOTPOP_CY}"
+        }
+      });
+
+      searchWidget.sources = sources;
+    ```
+
+5. In JSBin, run the app and type in "Downtown" or "St. John's". The app should highlight and zoom into the neighborhood polygon, and a popup should also be displayed.
+
+Your app should look something like this:
+* [Code](index.html)
+* [Live App](http://esri.github.io/geodev-hackerlabs/develop/jsapi3/search_with_widget/index.html)
+
+###Bonus
+* Add the default world geocoder to the search widget as well.
+* Add a [Rail Stops feature layer](http://services.arcgis.com/uCXeTVveQzP4IIcx/arcgis/rest/services/PDX_Rail_Stops_Styled/FeatureServer/0) to the Search widget.
+* Customize the popup's CSS. Hint: see the [Search with customization](https://developers.arcgis.com/javascript/jssamples/search_customized.html) sample.
```

Comment:
This link is to the 3.x documentation. Is there a 4.0 example? If not, probably remove this bonus section.